addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanner.java

```diff
@@ -0,0 +1,147 @@
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+/** Detect missing attribute integrity in supported elements */
+public class SubResourceIntegrityAttributeScanner extends PluginPassiveScanner {
+    /** Prefix for internationalized messages used by this rule */
+    private static final String MESSAGE_PREFIX = "pscanalpha.sri-integrity.";
+
+    private PassiveScanThread parent;
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+        // do nothing
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+
+        List<Element> sourceElements = source.getAllElements();
+        sourceElements.stream()
+                .filter(element -> SupportedElements.contains(element.getName()))
+                .filter(unsafeSubResource(msg.getRequestHeader().getHostName()))
+                .forEach(
+                        element -> {
+                            Alert alert =
+                                    new Alert(
+                                            getPluginId(),
+                                            Alert.RISK_MEDIUM,
+                                            Alert.CONFIDENCE_HIGH,
+                                            getName());
+
+                            alert.setDetail(
+                                    getString("desc"),
+                                    msg.getRequestHeader().getURI().toString(),
+                                    "",
+                                    "",
+                                    "",
+                                    getString("soln"),
+                                    getString("refs"),
+                                    element.toString(),
+                                    16, // CWE CATEGORY: Configuration
+                                    15, // Application Misconfiguration
+                                    msg);
+                            parent.raiseAlert(id, alert);
+                        });
+    }
+
+    private static Predicate<Element> unsafeSubResource(String origin) {
+        return element -> {
+            Optional<String> maybeHostname = SupportedElements.getHost(element);
+            return element.getAttributeValue("integrity") == null
+                    && !maybeHostname.map(hostname -> hostname.matches(origin)).orElse(false);
+        };
+    }
+
+    @Override
+    public void setParent(PassiveScanThread parent) {
+        this.parent = parent;
+    }
+
+    @Override
+    public String getName() {
+        return getString("name");
+    }
+
+    private static String getString(String param) {
+        return Constant.messages.getString(MESSAGE_PREFIX + param);
+    }
+
+    @Override
+    public int getPluginId() {
+        return 90003;
+    }
+
+    private enum SupportedElements {
```

Comment:
Any reason why this was moved to the end? It's easier to follow/read if it's at the beginning.